### PR TITLE
#122 - Added Headers abstraction

### LINKED ISSUE #122: Headers abstarction
HTTP headers are represented by type `Iterable<Map.Entry<String, String>>`. This is very verbose  type description and it would be better to have at least an alias and a point for possible future extension (ex. find all header values with same name, find header value with case-insensitive name)

### PR DESCRIPTION
**Pull request by olegmoz**
Closes #122 
Added `Headers ` interface, `Headers.From` for creating simple instances and their usage

### CODE REVIEW

**Comment by olegmoz** [comment]
@g4s8 no reviewer assigned for 18 hours, please look into this

File: src/main/java/com/artipie/http/Connection.java [hunk 1]
```diff
@@ -42,8 +41,5 @@
      * @param body The http response body.
      * @return Completion stage for accepting HTTP response.
      */
-    CompletionStage<Void> accept(
```
Comment (g4s8) [inline src/main/java/com/artipie/http/Connection.java:45]: @olegmoz maybe we can mark this method `@Deprecated` and convert to default interface method which calls new `accept(...)` with headers object?

Comment (olegmoz) [reply src/main/java/com/artipie/http/Connection.java:45]: @g4s8 fixed that, thanks

**Comment by 0crat** [comment]
This pull request [#123](https://github.com/artipie/http/issues/123) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/cf8b41fe-820a-4318-8dab-9f4de1745b29); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) I see that you unassigned this issue; the order is still assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk) though; to cancel the order use `refuse`, as in [§6](http://www.zerocracy.com/policy.html#6)

**Comment by 0crat** [comment]
Job [#123](https://github.com/artipie/http/issues/123) is closed, can't start order

**Comment by 0crat** [comment]
Job was finished in 0 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality bad

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31)

**Pull request merged**